### PR TITLE
APIO-310 expects property is missing from operations when requesting JSON-LD

### PR DESCRIPTION
**Pull request by ahdezma**
ℹ️ Steps to reproduce:
1) Request a resource with known operations

✅ Any {{Operation}} inside the {{operation}} field should contain an {{expects}} field pointing to the expected resource.
❌ The field is not present

```json
[
    {
        "target": "http://localhost:8080/o/api/commerce-web-site/37837/commerce-product-definition",
        "method": "POST",
        "@id": "_:commerce-web-site/commerce-product-definition/create",
        "@type": [
            "CreateAction",
            "Operation"
        ]
    }
]
```

### CODE REVIEW

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/action/ActionSemantics.java [hunk 1]
```diff
@@ -525,25 +549,26 @@ public FinalStep executeFunction(
...
+		 * <p>
+		 * If the action does not need information from the body, this method
+		 * shouldn't be called.
+		 * </p>
 		 *
 		 * @review
 		 */
-		public ActionSemantics build();
+		public FinalStep form(
+			Form form, Function2<Form, Body, Object> biFunction);
```
Comment (nhpatt) [inline apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/action/ActionSemantics.java]: function2 to be consistent with the previous method

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/action/Predicates.java [hunk 2]
```diff
@@ -220,7 +220,23 @@
...
+		return isActionForAny(resource);
+	}
+
+	/**
+	 * Returns a predicate that checks if the action's resource is equals to any
+	 * of the provided.
+	 *
+	 * @review
+	 */
+	public static Predicate<ActionSemantics> isActionForAny(
```
Comment (nhpatt) [inline apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/action/Predicates.java]: isActionForAnyPredicate :(

Comment (nhpatt) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/action/Predicates.java]: I think*

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/jaxrs/resource/FormResource.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	@Path("u/{name}")
+	public Form updaterForm(@PathParam("name") String name) {
+		return _getFormFromActionSemanticsMatching(
+			isReplaceAction.and(isActionFor(Item.of(name))));
+	}
+
+	@Reference
+	protected ActionManager actionManager;
+
+	private Form _getFormFromActionSemanticsMatching(
```
Comment (nhpatt) [inline apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/jaxrs/resource/FormResource.java]: get__Form

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/jaxrs/writer/FormMessageBodyWriter.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+		formFields.forEach(
+			formField -> {
+				Form form = formField.getForm();
+
+				if (form == null) {
+					jsonArray.put(toJSONObject(formField, null));
+				}
+				else {
+					JSONObject value = toJSONObject(
```
Comment (nhpatt) [inline apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/jaxrs/writer/FormMessageBodyWriter.java]: jsonObject

File: apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/jaxrs/writer/FormMessageBodyWriterTest.java [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+				"writeable", true
+			).put(
+				"value", nestedJSONObject
+			));
+
+		assertThat(jsonObject, is(sameJSONObjectAs(expected)));
+	}
+
+	private static JSONObject _createJSONObject(
+		Function1<Map<String, Object>, Map<String, Object>> function) {
```
Comment (nhpatt) [inline apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/jaxrs/writer/FormMessageBodyWriterTest.java]: :(

File: apio-architect-test/src/main/java/com/liferay/apio/architect/internal/test/jaxrs/resource/FormResourceTest.java [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+		assertThat(asJSONObject(response), is(sameJSONObjectAs(_404)));
+	}
+
+	@Test
+	public void testCreatorFormReturnsValidJSONObject() {
+		Response response = _makeRequestTo("f/c/name");
+
+		assertThat(response.getStatus(), is(200));
+
+		JSONObject expected = _getExpectedJSONObject(
```
Comment (nhpatt) [inline apio-architect-test/src/main/java/com/liferay/apio/architect/internal/test/jaxrs/resource/FormResourceTest.java]: :(

File: apio-architect-test/src/main/java/com/liferay/apio/architect/internal/test/jaxrs/resource/FormResourceTest.java [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+		assertThat(asJSONObject(response), is(sameJSONObjectAs(_404)));
+	}
+
+	@Test
+	public void testNestedCreatorFormReturnsValidJSONObject() {
+		Response response = _makeRequestTo("f/c/parent/name");
+
+		assertThat(response.getStatus(), is(200));
+
+		JSONObject expected = _getExpectedJSONObject(
```
Comment (nhpatt) [inline apio-architect-test/src/main/java/com/liferay/apio/architect/internal/test/jaxrs/resource/FormResourceTest.java]: :(

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/action/ActionSemantics.java (hunk 1 above)
Comment (ahdezma) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/action/ActionSemantics.java]: Bad refactor, thanks :)

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/jaxrs/resource/FormResource.java (hunk 3 above)
Comment (ahdezma) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/jaxrs/resource/FormResource.java]: Got it! Thanks :)

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/jaxrs/writer/FormMessageBodyWriter.java (hunk 4 above)
Comment (ahdezma) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/jaxrs/writer/FormMessageBodyWriter.java]: ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌@nhpatt‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ Wrong variable names
‌‌ ‌‌ ‌‌‌‌ ‌‌ ‌‌  ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌⬇️           ‌‌ ‌‌‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌  ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ⬇️

File: apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/jaxrs/writer/FormMessageBodyWriterTest.java (hunk 5 above)
Comment (ahdezma) [reply apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/jaxrs/writer/FormMessageBodyWriterTest.java]: :)

File: apio-architect-test/src/main/java/com/liferay/apio/architect/internal/test/jaxrs/resource/FormResourceTest.java (hunk 6 above)
Comment (ahdezma) [reply apio-architect-test/src/main/java/com/liferay/apio/architect/internal/test/jaxrs/resource/FormResourceTest.java]: Done!

File: apio-architect-test/src/main/java/com/liferay/apio/architect/internal/test/jaxrs/resource/FormResourceTest.java (hunk 7 above)
Comment (ahdezma) [reply apio-architect-test/src/main/java/com/liferay/apio/architect/internal/test/jaxrs/resource/FormResourceTest.java]: Got it :)

**Pull request merged**